internal/ethapi/api.go

```diff
@@ -1928,6 +1929,46 @@ func (s *PublicTransactionPoolAPI) SendRawTransaction(ctx context.Context, input
 	return SubmitTransaction(ctx, s.b, tx)
 }
 
+// SendRawTransactionConditional will add the signed transaction to the transaction pool.
+// The sender/bundler is responsible for signing the transaction
+func (s *PublicTransactionPoolAPI) SendRawTransactionConditional(ctx context.Context, input hexutil.Bytes, knownAccounts map[string]map[common.Address]interface{}) (common.Hash, error) {
+	tx := new(types.Transaction)
+	if err := tx.UnmarshalBinary(input); err != nil {
+		return common.Hash{}, err
+	}
+
+	if knownAccounts["knownAccounts"] == nil {
+		// should we allow this?
+		// throwing an error, for now
+		return common.Hash{}, errors.New("knownAccounts is empty")
+	}
```

Comment:
I don't think we should assume this to be always non-empty. Bundlers should be fully responsible for this field.

Reply:
Okay, that makes sense.